build.go

```diff
@@ -66,23 +70,63 @@ func Build(buildpackYMLParser BuildpackConfigParser, configParser ConfigParser,
 					{
 						Type:    "web",
 						Command: "watchexec",
-						Args:    []string{"--restart", "--shell", "sh", "--watch", context.WorkingDir, strconv.Quote(command)},
+						Args: append([]string{
+							"--restart",
+							"--watch", context.WorkingDir,
+							"--shell", "none",
+							"--",
+							command,
+						}, args...),
 						Default: true,
+						Direct:  true,
 					},
 					{
 						Type:    "no-reload",
 						Command: command,
+						Args:    args,
+						Direct:  true,
 					},
 				}
 			}
 		}
 
 		logger.LaunchProcesses(processes)
 
+		helperLayer, err := buildHelperLayer(context)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
 		return packit.BuildResult{
+			Layers: []packit.Layer{
+				helperLayer,
+			},
 			Launch: packit.LaunchMetadata{
 				Processes: processes,
 			},
 		}, nil
 	}
 }
+
+func buildHelperLayer(context packit.BuildContext) (helperLayer packit.Layer, err error) {
+	helperLayer, err = context.Layers.Get("dotnetexecute_helper")
+	helperLayer.Launch = true
+	if err != nil {
+		return packit.Layer{}, err
+	}
+
+	in := filepath.Join(context.CNBPath, "bin", "helper")
+	execdDir := filepath.Join(helperLayer.Path, "exec.d")
+	err = os.MkdirAll(execdDir, os.ModePerm)
+	if err != nil {
+		return packit.Layer{}, err
+	}
+	out := filepath.Join(execdDir, "helper")
+
+	err = fs.Copy(in, out)
+	if err != nil {
+		return packit.Layer{}, err
+	}
+
+	return
+}
```

Comment:
Generally speaking I am kinda of opposed to just having a private function like this. I see no reason why this shouldn't just be part of the main function in this format.

Reply:
I'll inline it